src/activity_handlers.cpp

```diff
@@ -2826,6 +2830,59 @@ void activity_handlers::wait_weather_finish( player_activity *act, player *p )
     act->set_to_null();
 }
 
+void activity_handlers::find_mount_do_turn( player_activity *act, player *p )
+{
+    //npc only activity
+    if( p->is_player() ) {
+        act->set_to_null();
+        return;
+    }
+    npc &guy = dynamic_cast<npc &>( *p );
+    monster *mon = guy.chosen_mount.lock().get();
+    if( !mon ) {
+        act->set_to_null();
+        guy.revert_after_activity();
+        return;
+    }
+    if( rl_dist( guy.pos(), mon->pos() ) <= 1 ) {
+        if( mon->has_effect( effect_controlled ) ) {
+            mon->remove_effect( effect_controlled );
+        }
+        if( p->can_mount( *mon ) ) {
+            act->set_to_null();
+            guy.revert_after_activity();
+            guy.chosen_mount = std::weak_ptr<monster>();
+            p->mount_creature( *mon );
+        } else {
+            if( p->get_skill_level( skill_survival ) < 1 ) {
```

Comment:
Checking this here instead of when the NPC chooses the mount leads to some funny behavior:

The NPC will walk right up to the chosen mount and when they read it, they suddenly recognize "Ups, I'm to large for this." or "Now that I'm close to it, I recognize, I don't known how to rid this."

This should probably be with the code that selects the mount, which should also have a check for reachability. No point in choosing a mount that is locked away.

Reply:
There was already a check for can_mount() in the npc_talk_func to find a mount.

I've added a reachability check to see if there is a path there.

Because of these checks, the potential message spam for the reason as to why the NPC cannot mount the horse is un-needed so ive removed it.